app/components-react/windows/SourceProperties.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useMemo, useState } from 'react';
+import { Services } from '../service-provider';
+import { ObsForm } from '../obs/ObsForm';
+import { TObsFormData } from '../../components/obs/inputs/ObsInput';
+import { useSelector } from '../store';
+import { ModalLayout } from '../shared/ModalLayout';
+import Display from '../shared/Display';
+import { assertIsDefined } from '../../util/properties-type-guards';
+
+export default function SourceProperties() {
+  const { WindowsService, SourcesService } = Services;
+
+  // get source
+  const source = useMemo(() => {
+    const { sourceId } = WindowsService.getChildWindowQueryParams();
+    return SourcesService.views.getSource(sourceId);
+  }, []);
+
+  // define reactive variables
+  const [properties, setProperties] = useState(source ? source.getPropertiesFormData() : []);
+  const hideStyleBlockers = useSelector(() => WindowsService.state.child.hideStyleBlockers);
+
+  function saveProperties(formData: TObsFormData) {
+    assertIsDefined(source);
+    // save source settings in OBS source
+    source.setPropertiesFormData(formData);
```

Comment:
This needs to go through the EditorCommandsService as source properties change is an undoable action.

Reply:
`EditSourcePropertiesCommand` is the name of the command